packages/ui/src/services/icon-config/icon-config.ts

```diff
@@ -1,73 +1,43 @@
-import { IconConfigPreset } from './types/preset'
-import { IconsConfig } from './types'
-import IconProps from './types/icon-props'
+import { IconsConfig, IconConfig } from './types'
+import { getRegexGroups } from './utils'
 
-export type IconMixinIcon = {
-  iconClass?: string
-  content?: string
-  component?: any
-  componentProps?: any
-} & IconProps
+function executeConfigFunctions (name: string, iconConfig: IconConfig): IconConfig {
+  const config = { ...iconConfig } as any
+  const args = getRegexGroups(name, config.name);
 
-function defaultPropsFromPreset (preset?: IconConfigPreset): IconProps {
-  if (!preset) { return {} }
-  return ['tag', 'color', 'rotation', 'spin'].map((key: string) => preset[key])
-}
+  ['iconClass', 'content', 'componentOptions'].forEach((funName) => {
+    if (typeof config[funName] === 'function') {
+      config[funName] = config[funName](...args)
+    }
+  })
 
-function getRegexGroups (str: string, regex: RegExp) {
-  return str.match(regex)?.slice(1)
+  return config
 }
 
-function getCssIcon (name: string, preset?: IconConfigPreset) {
-  if (preset?.type === 'css') {
-    const regexGroups = getRegexGroups(name, preset.name)
+function findConfigByName (name: string, iconsConfig: IconsConfig, ignoreNames: string[] = []): IconConfig | undefined {
+  const config = iconsConfig.find((c) => {
+    if (ignoreNames.includes(c.name.toString())) { return false }
+    return typeof c.name === 'string' ? c.name === name : c.name.test(name)
+  })
 
-    if (regexGroups) {
-      return { iconClass: preset.iconClass(...regexGroups) }
-    } else {
-      return { iconClass: name }
-    }
+  if (config?.to) {
+    const preset = findConfigByName(config.to, iconsConfig, [...ignoreNames, name])
+    if (preset) { return { ...preset, ...config, name: preset.name } }
   }
-}
-
-function getLigatureIcon (name: string, preset?: IconConfigPreset) {
-  if (preset?.type === 'ligature') {
-    const regexGroups = getRegexGroups(name, preset.name)
 
-    if (regexGroups) {
-      return { content: preset.content(...regexGroups), iconClass: preset.iconClass }
-    } else {
-      return { content: name, iconClass: preset.iconClass }
-    }
-  }
+  return config
 }
 
-function getComponentIcon (name: string, preset?: IconConfigPreset) {
-  if (preset && preset.type === 'component') {
-    const regexGroups = getRegexGroups(name, preset.name)
-
-    if (regexGroups) {
-      return {
-        componentProps: preset.componentProps(...regexGroups),
-        component: preset.component,
-      }
-    } else {
-      return { component: preset.component }
-    }
+export function getIconConfig (name: string, iconsConfig?: IconsConfig): IconConfig {
+  if (!iconsConfig) {
+    throw new Error(`Cant find config for name="${name}". You can create default config. Visit DOCS // TODO`)
```

Comment:
Probably we don't need the check here. Having icon config is guaranteed.

Reply:
According to config types value can be `undefined`. User can pass `undefined` by accident, so we need to check this.